### PR TITLE
OLM Backport with securityContext field added for 2.8

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
OLM Backport with `securityContext` field added for 2.8.0 release

### CODE REVIEW

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 1]
```diff
@@ -362,25 +362,22 @@ metadata:
...
     operators.operatorframework.io/internal-objects: |-
       ["strimzipodsets.core.strimzi.io"]
     olm.skipRange: '>=2.7.0-0 <2.8.0-0'
+    features.operators.openshift.io/disconnected: “true”
+    features.operators.openshift.io/fips-compliant: “true”
+    features.operators.openshift.io/proxy-aware: “true”
+    features.operators.openshift.io/tls-profiles: “false”
+    features.operators.openshift.io/token-auth-aws: “false”
+    features.operators.openshift.io/token-auth-azure: “false”
+    features.operators.openshift.io/token-auth-gcp: “false”
```
Comment (ppatierno) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:380]: why this section was moved down and we missed some of the the fields?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 2]
```diff
@@ -1190,7 +1206,7 @@ spec:
           replicas: 1
           selector:
             matchLabels:
-              name: amq-streams-cluster-operator
+              name: strimzi-cluster-operator
```
Comment (ppatierno) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1209]: We have `amq-streams-cluster-operator` in all previous releases, or?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 3]
```diff
@@ -1199,10 +1215,10 @@ spec:
...
                   3.7.0=registry.redhat.io/amq-streams/kafka-37-rhel9:2.8.0-0
                   3.8.0=registry.redhat.io/amq-streams/kafka-38-rhel9:2.8.0-0
-                kafka-previous-image: registry.redhat.io/amq-streams/kafka-37-rhel9:2.8.0-0
-                kafka-current-image: registry.redhat.io/amq-streams/kafka-38-rhel9:2.8.0-0
-                bridge-image: registry.redhat.io/amq-streams/bridge-rhel9:2.8.0-0
-                maven-image: registry.redhat.io/amq-streams/maven-builder-rhel9:2.8.0-0
+                  kafka-previous-image: registry.redhat.io/amq-streams/kafka-37-rhel9:2.8.0-0
+                  kafka-current-image: registry.redhat.io/amq-streams/kafka-38-rhel9:2.8.0-0
+                  bridge-image: registry.redhat.io/amq-streams/bridge-rhel9:2.8.0-0
+                  maven-image: registry.redhat.io/amq-streams/maven-builder-rhel9:2.8.0-0
```
Comment (ppatierno) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1221]: not sure having this indentation is right.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 4]
```diff
@@ -1246,8 +1264,8 @@ spec:
                       value: "300000"
                     - name: STRIMZI_DEFAULT_TLS_SIDECAR_ENTITY_OPERATOR_IMAGE
                       valueFrom:
-                        fieldRef:
-                          fieldPath: metadata.annotations['kafka-current-image']
+                      fieldRef:
+                      fieldPath: metadata.annotations['kafka-current-image']
```
Comment (ppatierno) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1268]: wrong indentation

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 5]
```diff
@@ -1429,7 +1447,7 @@ spec:
   maturity: stable
   provider:
     name: Red Hat
-  replaces: amqstreams.v2.7.0-6
+  replaces: strimzi-cluster-operator.v0.42.0
```
Comment (ppatierno) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1450]: wrong

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 6]
```diff
@@ -905,7 +921,7 @@ spec:
...
...bGFzcz0iY2xzLTQiIGQ9Ik0yMC4xMiwxNS41YzAtMS43Mi0xLjQtMy4xMi0zLjEyLTMuMTJzLTMuMTIsMS40LTMuMTIsMy4xMiwxLjQsMy4xMiwzLjEyLDMuMTIsMy4xMi0xLjQsMy4xMi0zLjEyWk0xNywxNy4zOGMtMS4wMywwLTEuODgtLjg0LTEuODgtMS44OHMuODQtMS44OCwxLjg4LTEuODgsMS44OC44NCwxLjg4LDEuODgtLjg0LDEuODgtMS44OCwxLjg4WiIvPjxwYXRoIGNsYXNzPSJjbHMtNCIgZD0iTTIzLDE4LjYyYy4zNCwwLC42Mi0uMjguNjItLjYydi01YzAtLjM1LS4yOC0uNjItLjYyLS42MnMtLjYyLjI4LS42Mi42MnY1YzAsLjM1LjI4LjYyLjYyLjYyWiIvPjxwYXRoIGNsYXNzPSJjbHMtNCIgZD0iTTIxLDE5LjM4Yy0xLjcyLDAtMy4xMiwxLjQtMy4xMiwzLjEyczEuNCwzLjEyLDMuMTIsMy4xMiwzLjEyLTEuNCwzLjEyLTMuMTItMS40LTMuMTItMy4xMi0zLjEyWk0yMSwyNC4zOGMtMS4wMywwLTEuODgtLjg0LTEuODgtMS44OHMuODQtMS44OCwxLjg4LTEuODgsMS44OC44NCwxLjg4LDEuODgtLjg0LDEuODgtMS44OCwxLjg4WiIvPjxwYXRoIGNsYXNzPSJjbHMtNCIgZD0iTTE1LDE5LjM4Yy0uMzUsMC0uNjIuMjgtLjYyLjYydjVjMCwuMzUuMjguNjIuNjIuNjJzLjYyLS4yOC42Mi0uNjJ2LTVjMC0uMzUtLjI4LS42Mi0uNjItLjYyWiIvPjxwYXRoIGNsYXNzPSJjbHMtNCIgZD0iTTI3LDEyLjM4Yy0uMzQsMC0uNjIuMjgtLjYyLjYydjVjMCwuMzUuMjguNjIuNjIuNjJzLjYyLS4yOC42Mi0uNjJ2LTVjMC0uMzUtLjI4LS42Mi0uNjItLjYyWiIvPjxwYXRoIGNsYXNzPSJjbHMtNCIgZD0iTTExLDE5LjM4Yy0uMzUsMC0uNjIuMjgtLjYyLjYydjVjMCwuMzUuMjguNjIuNjIuNjJzLjYyLS4yOC42Mi0uNjJ2LTVjMC0uMzUtLjI4LS42Mi0uNjItLjYyWiIvPjwvZz48L3N2Zz4=
```
Comment (kyguy) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:908]: We want to keep the original image here

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 7]
```diff
@@ -838,23 +835,42 @@ spec:
...
+    * Specifying additional volumes and volume mounts in Streams custom resources
+
+    * Support for custom Cruise Control API users
+
+    * Unregistration of KRaft nodes after scale-down
+
+    * Kafka Exporter updated to 1.8.0 (and the Grafana dashboard updated to work with it)
+
+    * Service account based authentication
+
```
Comment (kyguy) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:873]: We can remove this block, it is specific to 0.43.0 from what I understand, not the range of 0.39.0-0.43.0

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 3 above)
Comment (kyguy) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1221]: Keep the indentation as it was here

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 8]
```diff
@@ -362,25 +362,22 @@ metadata:
...
-    features.operators.openshift.io/fips-compliant: "true"
-    features.operators.openshift.io/cnf: "false"
-    features.operators.openshift.io/cni: "false"
-    features.operators.openshift.io/csi: "false"
-    features.operators.openshift.io/tls-profiles: "false"
-    features.operators.openshift.io/token-auth-aws: "false"
-    features.operators.openshift.io/token-auth-azure: "false"
-    features.operators.openshift.io/token-auth-gcp: "false"
-    operators.openshift.io/valid-subscription: |-
-      ["Red Hat AMQ", "Red Hat Application Foundations"]
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:380]: Correct ones added back

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 1 above)
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Moved back and missing fields returned

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 2 above)
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Correcy. Now fixed

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 3 above)
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Fixed

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 4 above)
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Fixed

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 7 above)
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Removed

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 9]
```diff
@@ -376,6 +376,8 @@ metadata:
     features.operators.openshift.io/token-auth-aws: "false"
     features.operators.openshift.io/token-auth-azure: "false"
     features.operators.openshift.io/token-auth-gcp: "false"
+    operators.openshift.io/infrastructure-features: |-
```
Comment (kyguy) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:379]: This is not need, it should be covered by the annotations above it

Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:379]: It shouldn't cause any issues though

**Review by kyguy — APPROVED** [review]

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 9 above)
Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:379]: @kyguy I get an error when I remove it as it's covering `["disconnected", "proxy", "proxy-aware", "fips"]`

Comment (OwenCorrigan76) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Removed

**Review by ppatierno — APPROVED** [review]

**Pull request merged**